e2e-identity/tests/utils/docker/dex.rs

```diff
@@ -47,36 +49,36 @@ impl DexImage {
 
         std::fs::write(&host_cfg_file, cfg.to_yaml(redirect_uri)).unwrap();
 
-        let host_vol_str = host_cfg_file.as_os_str().to_str().unwrap().to_string();
+        let host_vol_str = host_cfg_file.as_os_str().to_str().unwrap();
         Self {
-            volumes: HashMap::from_iter(vec![(host_vol_str, "/etc/dex/config.docker.yaml".to_string())]),
+            volumes: vec![Mount::bind_mount(host_vol_str, "/etc/dex/config.docker.yaml")],
             env_vars: HashMap::new(),
         }
     }
 }
 
 impl Image for DexImage {
-    type Args = ();
-
-    fn name(&self) -> String {
-        Self::NAME.to_string()
+    fn name(&self) -> &str {
+        Self::NAME
     }
 
-    fn tag(&self) -> String {
-        std::env::var("DEX_VERSION").unwrap_or_else(|_| Self::TAG.to_string())
+    fn tag(&self) -> &str {
+        std::env::var("DEX_VERSION")
+            .map(|tag| Box::leak(tag.into_boxed_str()) as &str)
+            .unwrap_or(Self::TAG)
```

Comment:
Why leak here? Why don't we get the tag at `DexImage` init time and then just give out a proper reference?